websites/B/bilibilitv/presence.ts

```diff
@@ -1,97 +1,164 @@
+			)?.textContent ||
+			document.querySelector(
+				"div.layout-normal > div.layout-body.media-width > div > ul > li:nth-child(2) > a"
+			)?.textContent,
+		playing = document.querySelector(
+			"img.player-mobile-icon.player-mobile-pause-icon.player-mobile-active"
+		);
+	if (oldLang !== newLang || !strings) {
+		oldLang = newLang;
+		strings = await getStrings();
+	}
 	// Main Site
-
-	if (path === "/en") {
-		presenceData.smallImageKey = "idle";
-		presenceData.details = "Currently browsing";
-		presenceData.state = "Homepage";
-	} else if (path.startsWith("/en/play")) {
-		presenceData.smallImageKey = "play";
-		presenceData.smallImageText = "Watching a show";
-		presenceData.details = getElement("h1.video-info__title");
-		presenceData.state = `Episode ${getEpisode("title")}`;
-		presenceData.startTimestamp = browsingStamp;
-		presenceData.buttons = [
-			{
-				label: "Watch this show",
-				url: `https://www.bilibili.tv/en/play/${urlpath[3]}`
+	if (document.location.hostname === "www.bilibili.tv") {
+		switch (document.location.pathname.toString().toLowerCase().split("/")[2]) {
+			case "video": {
+				presenceData.details = strings.watchingVid;
+				presenceData.state = title;
+				presenceData.startTimestamp = browsingStamp;
+				presenceData.smallImageKey = playing ? "pause" : "play";
+				presenceData.smallImageText = playing ? strings.play : strings.pause;
+				presenceData.largeImageKey = document
+					.querySelector<HTMLImageElement>('meta[name="og:image"]')
+					.getAttribute("content")
+					.split("@")[0];
+				presenceData.buttons = [
+					{
+						label: strings.buttonWatchVideo,
+						url: document
+							.querySelector('meta[name="og:url"]')
+							.getAttribute("content")
+					}
+				];
+				break;
 			}
-		];
-	} else if (path.startsWith("/en/video")) {
-		presenceData.smallImageKey = "play";
-		presenceData.smallImageText = "Watching a video";
-		presenceData.details = getElement("h1.video-info__title");
-		presenceData.state = `by ${getElement(".video-info__creator--nickname")}`;
-		presenceData.startTimestamp = browsingStamp;
-		presenceData.buttons = [
-			{
-				label: "Watch this video",
-				url: `https://www.bilibili.tv/en/video/${urlpath[3]}`
+			case "play": {
+				let ep;
+				if (
+					document.querySelector("span.series-text.active") &&
+					!document
+						.querySelector("div.video-info__title-wrap > h1 > a")
+						.textContent.includes(
+							document.querySelector("span.series-text.active").textContent
+						)
+				) {
+					ep = `${document
+						.querySelector("span.series-text.active")
+						.textContent.replace(/ *\([^)]*\) */g, "")} | ${strings.episode} ${
+						title.match(/\d+/g)[0]
+					}`;
+				} else ep = `${strings.episode} ${title.match(/\d+/g)[0]}`;
+				presenceData.details = `${
+					document.querySelector("div.video-info__title-wrap > h1 > a")
+						.textContent
+				}`;
+				presenceData.state = `${ep}`;
```

Comment:
no need to stringify strings
```suggestion
				presenceData.state = ep;
```